driver-core/src/main/com/mongodb/client/model/Filters.java

```diff
@@ -247,8 +367,51 @@ public static Bson or(final Bson... filters) {
         return or(asList(filters));
     }
 
-    // TODO: $not
-    // TODO: $nor
+    /**
+     * Creates a filter that matches all documents where the value of the field name does not match the specified value.
+     *
+     * @param fieldName the field name
+     * @param value     the value
+     * @return the filter
+     * @mongodb.driver.manual reference/operator/query/not $not
+     */
+    public static Bson not(final String fieldName, final Bson value) {
+        return new OperatorFilter<Bson>("$not", fieldName, value);
+    }
+
+    /**
+     * Creates a filter that matches all documents where the value of the field name does not match the specified value.
+     *
+     * @param fieldName the field name
+     * @param value     the value
+     * @return the filter
+     * @mongodb.driver.manual reference/operator/query/not $not
+     */
+    public static Bson not(final String fieldName, final Pattern value) {
```

Comment:
This is the only place in Filters where we have to use something besides Bson as a value.